test/libyul/yulOptimizerTests/fullSuite/remove_redundant_assignments_in_switch.yul

```diff
@@ -11,9 +11,10 @@
 //
 // {
 //     {
+//         let x := 0
 //         switch mload(0)
-//         case 0 { }
-//         case 1 { }
+//         case 0 { x := 0 }
+//         case 1 { x := 1 }
 //         default { invalid() }
```

Comment:
This one also seems to be the fault of the removed simplification segment. Adding an extra `u` solves both this and the unused parameter pruner regression (https://github.com/ethereum/solidity/pull/14928/#r1541097978).